src/Boleto.java

```diff
@@ -1,43 +1,85 @@
 public class Boleto {
-    int precioPelicula;
+    double precioPelicula;
     int cantidadAsientos;
     int puntosCliente = 50;
     Cliente cliente;
     Sala salaAsignada;
     String generoPelicula;
 
-    public Boleto(Cliente cliente, int precioPelicula, Sala sala){
+    String tipoPelicula;
+
+    public Boleto(Cliente cliente, double precioPelicula, Sala salaAsignada, String tipoPelicula){
         this.cliente = cliente;
         this.precioPelicula = precioPelicula;
-        this.salaAsignada = sala;
+        this.salaAsignada = salaAsignada;
+        this.tipoPelicula=tipoPelicula;
     }
 
     public void setGeneroPelicula(String generoPelicula) {
         this.generoPelicula = generoPelicula;
     }
 
-    public void aplicarDescuento(Boleteria.MetodoPago metodo){
+    public void aplicarDescuento(Boleteria.MetodoPago metodo,Semana dia,String tipoPelicula){
```

Comment:
En esta funcion no es necesario clasificar el tipo de pelicula, la clase boleteria te pasa el precio clasificado de antemano. 
```suggestion
    public void aplicarDescuento(Boleteria.MetodoPago metodo,Semana dia){
```